learning/tour-of-beam/frontend/lib/pages/tour/state.dart

```diff
@@ -178,114 +176,79 @@ class TourNotifier extends ChangeNotifier with PageStateMixin<void> {
         snippetEditingController.activeFileController?.isChanged ?? false;
     final snippetFiles = snippetEditingController.getFiles();
 
-    final doSave = _authNotifier.isAuthenticated &&
-        isCodeChanged &&
-        currentUnitController != null &&
-        snippetFiles.isNotEmpty;
+    final doSave =
+        isCodeChanged && _currentUnitContent != null && snippetFiles.isNotEmpty;
 
     if (doSave) {
+      final sdk = currentSdk;
+      final unitId = currentUnitId;
```

Comment:
?

Reply:
It's a snapshot of sdk and unitId at the moment of editing. Added the comment.